core/src/main/java/org/apache/struts2/components/Include.java

```diff
@@ -90,27 +92,50 @@
 
     private static final Logger LOG = LogManager.getLogger(Include.class);
 
-    private static String systemEncoding = System.getProperty("file.encoding");
+    private static final String systemEncoding = System.getProperty("file.encoding");
 
     protected String value;
     private HttpServletRequest req;
     private HttpServletResponse res;
-    private static String defaultEncoding;
+    private String defaultEncoding;   // Made non-static (during WW-4971 fix)
+    private String overrideEncoding;  // Added with WW-4971 fix (allow direct encoding override via tag)
+    private boolean usePageEncoding;  // Added with WW-4971 fix (allow usage of page or default encoding)
 
     public Include(ValueStack stack, HttpServletRequest req, HttpServletResponse res) {
         super(stack);
         this.req = req;
         this.res = res;
+        defaultEncoding = "UTF-8";  // Set UTF-8 for defaultEncoding, when not set in configuration
```

Comment:
Why to override `defaultEncoding` in constructor? It will be set in `setDefaultEncoding`.

Reply:
Just a case of being extra defensive (in case `setDefaultEncoding()` injection parameter was ever changed to optional).  It's probably unlikely to occur, in which case that assignment in the constructor could be eliminated.